plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,288 @@
+	Exclude    struct {
+		Schemas []string `mapstructure:"schemas"`
+		Tables  []string `mapstructure:"tables"`
+	} `mapstructure:"exclude,omitempty"`
+	Concurrency      int  `mapstructure:"concurrency,omitempty"`
+	BuildViewLineage bool `mapstructure:"build_view_lineage,omitempty"`
+}
+
+type Extractor struct {
+	plugins.BaseExtractor
+	logger log.Logger
+	config Config
+}
+
+//go:embed README.md
+var summary string
+
+var sampleConfig = `
+project_name: goto_test
+endpoint_project: https://service.ap-southeast-5.maxcompute.aliyun.com/api
+access_key:
+    id: access_key_id
+    secret: access_key_secret
+schema_name: default
+exclude:
+	schemas:
+    	- schema_a
+    	- schema_b
+    tables:
+    	- schema_c.table_a
+concurrency: 10
+build_view_lineage: true
+`
+
+var info = plugins.Info{
+	Description:  "MaxCompute metadata and metrics",
+	SampleConfig: sampleConfig,
+	Tags:         []string{"alicloud", "table"},
+	Summary:      summary,
+}
+
+func New(logger log.Logger) *Extractor {
+	e := &Extractor{
+		logger: logger,
+	}
+	e.BaseExtractor = plugins.NewBaseExtractor(info, &e.config)
+	e.ScopeNotRequired = true
+
+	return e
+}
+
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
+	return e.BaseExtractor.Init(ctx, config)
+}
+
+func (e *Extractor) Extract(_ context.Context, emit plugins.Emit) error {
+	aliAccount := account.NewAliyunAccount(e.config.AccessKey.ID, e.config.AccessKey.Secret)
+	odpsIns := odps.NewOdps(aliAccount, e.config.EndpointProject)
+	project := odpsIns.Project(e.config.ProjectName)
+
+	schemas := project.Schemas()
+
+	eg := errgroup.Group{}
+
+	err := schemas.List(func(schema *odps.Schema, err error) {
+		if err != nil {
+			e.logger.Error("failed to list schemas", "error", err)
+			return
+		}
+		if e.config.SchemaName != "" && schema.Name() != e.config.SchemaName {
+			return
+		}
+		if contains(e.config.Exclude.Schemas, schema.Name()) {
+			return
+		}
+
+		newIns := odps.NewOdps(aliAccount, e.config.EndpointProject)
+		newIns.SetCurrentSchemaName(schema.Name())
+		newIns.SetDefaultProjectName(e.config.ProjectName)
+		newProj := newIns.Project(e.config.ProjectName)
+
+		eg.Go(func() error {
+			return e.fetchTablesFromSchema(newProj, emit)
+		})
+	})
+	if err != nil {
+		return err
+	}
+	return eg.Wait()
+}
+
+func (e *Extractor) fetchTablesFromSchema(project *odps.Project, emit plugins.Emit) error {
+	eg := errgroup.Group{}
+	project.Tables().List(
+		func(t *odps.Table, err error) {
+			if err != nil {
+				e.logger.Error("table list", err)
+				return
+			}
+
+			if contains(e.config.Exclude.Tables, fmt.Sprintf("%s.%s", t.SchemaName(), t.Name())) {
+				return
+			}
+
+			eg.Go(func() error {
+				return e.processTable(t, emit)
+			})
+		},
+	)
+	return eg.Wait()
+}
+
+func (e *Extractor) processTable(table *odps.Table, emit plugins.Emit) error {
+	err := table.Load()
+	if err != nil {
+		e.logger.Error("failed to get table info", "table", table.Name(), "error", err)
```

Comment:
at the moment there's an error when trying to load table with type view, while waiting Ali team response, let's handle this, as long as the table basic info loaded, we can ignore the [error coming from extended info](https://github.com/aliyun/aliyun-odps-go-sdk/blob/v0.3.14/odps/table.go#L152-L155)

```suggestion
		isView := table.Schema().IsVirtualView || table.Schema().IsMaterializedView
		isLoaded := table.IsLoaded()
		if !isView || (isView && !isLoaded) {
			e.logger.Error("failed to get table info", "schema", table.SchemaName(), "table", table.Name(), "error", err)
			return err
		}
```